spring-kafka-2/src/test/java/com/baeldung/spring/kafka/managingkafkaconsumergroups/ManagingConsumerGroupsIntegrationTest.java

```diff
@@ -0,0 +1,55 @@
+package com.baeldung.spring.kafka.managingkafkaconsumergroups;
+
+import org.apache.commons.lang3.RandomUtils;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.kafka.config.KafkaListenerEndpointRegistry;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.listener.MessageListenerContainer;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@SpringBootTest(classes = ManagingConsumerGroupsApplicationKafkaApp.class)
+@EmbeddedKafka(partitions = 2, brokerProperties = {"listeners=PLAINTEXT://localhost:9092", "port=9092"})
+public class ManagingConsumerGroupsIntegrationTest {
+
+    @Autowired
+    KafkaTemplate<String, Double> kafkaTemplate;
+
+    @Autowired
+    KafkaListenerEndpointRegistry kafkaListenerEndpointRegistry;
+
+    @Autowired
+    MessageConsumerService consumerService;
+
+    @Test
+    public void givenContinuousMessageFlow_whenAConsumerLeavesTheGroup_thenKafkaTriggersPartitionRebalance() throws InterruptedException {
+        int currentMessage = 0;
+        final String consumer1Identifier = "org.springframework.kafka.KafkaListenerEndpointContainer#1";
+        final int totalProducedMessages = 50000;
+        final int messageWhereConsumer1LeavesGroup = 10000;
```

Comment:
If there's no very specific reason we avoid the use of final in code (we all know that immutability objects have several benefits, but for demonstration code, we favor simplicity/brevity over immutability).

In fact, I'm thinking we can move those values out of the method and declare them as constant, what do you think? In that case, we reduce even more the code for the article.

Reply:
I liked the idea of global constants